src/add-text-track-data.js

```diff
@@ -64,21 +74,46 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
   }
 
   if (metadataArray) {
+    let videoDuration = durationOfVideo(sourceHandler.mediaSource_.duration);
+
     metadataArray.forEach(function(metadata) {
       let time = metadata.cueTime + this.timestampOffset;
 
       metadata.frames.forEach(function(frame) {
         let cue = new Cue(
-            time,
-            time,
-            frame.value || frame.url || frame.data || '');
+          time,
+          time,
+          frame.value || frame.url || frame.data || '');
 
         cue.frame = frame;
         cue.value = frame;
         deprecateOldCue(cue);
+
         this.metadataTrack_.addCue(cue);
       }, this);
     }, sourceHandler);
+
+    /** Updating the metadeta cues so that
+     * the endTime of each cue is the startTime of the next cue
+     * the endTime of last cue is the duration of the video
+     */
+    if (sourceHandler.metadataTrack_ && sourceHandler.metadataTrack_.cues) {
+      let cues = sourceHandler.metadataTrack_.cues;
+      let cuesArray = [];
+
+      for (let j = 0; j < cues.length; j++) {
+        cuesArray.push(cues[j]);
+      }
+      cuesArray.sort(function(first, second) {
```

Comment:
You can use an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) here.